KomaMRIBase/src/datatypes/phantom/motion/SimpleMotion.jl

```diff
@@ -71,7 +71,7 @@ function get_spin_coords(
     x::AbstractVector{T},
     y::AbstractVector{T},
     z::AbstractVector{T},
-    t::AbstractArray{T},
+    t::AbstractArray{T}
```

Comment:
That is for making copies.
`aux = xt, yt, zt` would overwrite these 3 arrays.

The equivalent to `aux = xt .+ 0, yt .+ 0, zt .+ 0`  would be:
`aux = copy(xt), copy(yt), copy(zt)`

Reply:
Isn't it the idea to do in-place operations after allocating `xt,` `yt`, and `zt`? Why generate a copy each time? 

This can be optimized.

Reply:
I don't get this

Reply:
```julia
...
# (Step 1) x, y, and z have size (Ns x 1)
# (Step 2) Then, we copy coords to xt, yt, and zt of size (Ns x Nt)
xt, yt, zt = x .+ 0*t, y .+ 0*t, z .+ 0*t
# (Step 3) We will update each coord (Ns x Nt) over time recursively
for motion in Iterators.filter(is_composable, motion.types)
	# (Step 4) We update each coord (Ns x Nt) without allocations
    xt, yt, zt  .+= displacements(motion, xt, yt, zt, t)
end
...

```
I remember that we had a problem doing this, but if it uses less memory, it could be worth doing.

Reply:
Apparently, this line is generating problems with Metal (@pvillacorta , @rkierulf ).

Reply:
I removed the extra allocations in [1315be2](https://github.com/JuliaHealth/KomaMRI.jl/pull/408/commits/1315be242aeb673398bfb60a8b100752072edb84). However, I don't know if this will be enough for solving tests fails with Metal, since the commit solves a problem composable motions, and the test only executes a Translation (non-composable) motion